client/src/commands/UserInputUtil.ts

```diff
@@ -445,4 +470,120 @@ export class UserInputUtil {
+                    data.add(peerName);
+                    quickPickItems.push({label: channel, data: data});
+                }
+            });
+        }
+        return quickPickItems;
+    }
+
+    private static async getInstalledContracts(connection: IFabricConnection, peers: Set<string>): Promise<IBlockchainQuickPickItem<{ packageEntry: PackageRegistryEntry, workspace: vscode.WorkspaceFolder }>[]> {
+        const tempQuickPickItems: IBlockchainQuickPickItem<{ packageEntry: PackageRegistryEntry, workspace: vscode.WorkspaceFolder }>[] = [];
+        for (const peer of peers) {
+            const chaincodes: Map<string, Array<string>> = await connection.getInstalledChaincode(peer);
+            chaincodes.forEach((versions: string[], chaincodeName: string) => {
+                versions.forEach((version: string) => {
+                    const _package: PackageRegistryEntry = {name: chaincodeName, version: version, path: undefined } as PackageRegistryEntry; // PATH SHOULD BE SET
+                    const data: {packageEntry: PackageRegistryEntry, workspace: vscode.WorkspaceFolder} = {packageEntry: _package, workspace: undefined};
+                    const label: string = `${chaincodeName}@${version}`;
+                    tempQuickPickItems.push({label: label, description: 'Installed', data: data});
+                });
+            });
+        }
+        return tempQuickPickItems;
+    }
+
+    private static async getPackagedContracts(tempQuickPickItems: IBlockchainQuickPickItem<{ packageEntry: PackageRegistryEntry, workspace: vscode.WorkspaceFolder }>[]): Promise<IBlockchainQuickPickItem<{ packageEntry: PackageRegistryEntry, workspace: vscode.WorkspaceFolder }>[]> {
+
+        const packages: Array<PackageRegistryEntry> = await PackageRegistry.instance().getAll();
+
+        for (const _package of packages) {
+            const data: {packageEntry: PackageRegistryEntry, workspace: vscode.WorkspaceFolder} = {packageEntry: _package, workspace: undefined};
+            const label: string = `${_package.name}@${_package.version}`;
+            const packageIsFound: number = tempQuickPickItems.findIndex((item: any) => {
+                return item.label === label;
+            });
+            if (packageIsFound === -1) {
+                tempQuickPickItems.push({label: label, description: 'Packaged', data: data});
+            }
+        }
+
+        return tempQuickPickItems;
+    }
+
+    private static async getOpenProjects(tempQuickPickItems: IBlockchainQuickPickItem<{ packageEntry: PackageRegistryEntry, workspace: vscode.WorkspaceFolder }>[]): Promise<IBlockchainQuickPickItem<{ packageEntry: PackageRegistryEntry, workspace: vscode.WorkspaceFolder }>[]> {
+
+        const workspaceFolderOptions: vscode.WorkspaceFolder[] = UserInputUtil.getWorkspaceFolders();
+        for (const workspace of workspaceFolderOptions) {
+
+            const packagePath: string = path.join(workspace.uri.fsPath, 'package.json');
```

Comment:
we do this in a few places maybe we should have a function in extensionUtils or another util class for this